dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/AppClassLoader.java

```diff
@@ -702,13 +701,28 @@ final ByteResourceInformation findClassBytes(String className, String resourceNa
      * Find a class on this loader's class path or delegate to the parent class
      * loader.
      */
+    @FFDCIgnore(ClassNotFoundException.class)
     protected Class<?> findOrDelegateLoadClass(String name) throws ClassNotFoundException {
-        // The resolve parameter is a legacy parameter that is effectively
-        // never used as of JDK 1.1 (see footnote 1 of section 5.3.2 of the 2nd
-        // edition of the JVM specification).  The only caller of
-        // loadClass(String, boolean) is java.lang.ClassLoader.loadClass(String),
-        // and that method always passes false, so we ignore the parameter.
-        return super.loadClass(name, false);
+        Class<?> result = findLoadedClass(name);
+        if (result != null) {
+            return result;
+        }
+        if (parent instanceof NoClassNotFoundLoader) {
+            result = ((NoClassNotFoundLoader) parent).loadClassNoException(name);
+        } else {
+            try {
+                if (parent == null) {
+                    return super.loadClass(name, false);
+                }
+                result = parent.loadClass(name);
+            } catch (ClassNotFoundException e) {
+                // move on to local findClass
+            }
+        }
+        if (result != null) {
+            return result;
+        }
+        return findClass(name);
```

Comment:
This has been wrong for a while it would appear.  findClass also does a findLoadedClass().  It appears that it shouldn't do that and should just look for the class without looking at the loaded classses.  Of course this will require and update to the ParentLastClassLoader to call findLoadedClass() before calling findClass() if we change the way that findClass() works.

Reply:
I think findClass currently must call findLoadedClass because direct calls to LibertyLoader (which AppClassLoader eventually extends) done in `AppClassLoader.findClassCommonLibraryClassLoaders(String)`

Reply:
ok I will take this under advisement when looking at doing some cleanup here in a follow one PR.